docs/authentication.md

```diff
@@ -0,0 +1,194 @@
+
+# Authenticating with the CE-API  
+  
+All read operations in the Contributor Environment API are publicly accessible. However, in order to create, update, or delete nodes, the request **MUST BE** authenticated with a JWT token.
+
+## Request a JWT token
+
+The Contributor Environment API has an endpoint to create a JWT token. This endpoint **MUST ONLY** be used in a secured environment since it uses a private API key which shouldn't be visible to users. Best practice for web applications is to request the JWT token in a backend environment and to only use the JWT token in the frontend.
+
+### JWT Endpoint
+
+For test environment use:
+
+`https://api-test.trompamusic.eu/jwt`
+
+For production environment use:
+
+`https://api.trompamusic.eu/jwt`
+
+> **NOTE:** API keys are not exchangeable between the test and production environment. Each environment will have its own unique API keys.
+
+### Parameters
+
+In order to obtain a JWT token you will need to have the following information:
+
+| Parameter | Type | Description |
+| -- | -- | -- |
+| id | String | This will identity your token (like a username) |
+| apiKey | String | This is your secret API key (like a password) |
+| scopes | String[] | A list of scopes for the required operations and types |
+
+> The `id` and `apiKey` can be requested from one of the TROMPA project partners.
+
+### Scopes
+
+When requesting a JWT token, you can determine to which resources the token has access to. For example, if you only need to create a Person in the client side code, the following scope will only allow the client side code to create a person: `"scopes": ["Mutation:CreatePerson"]` 
+
+The scope pattern uses the `Operation` and `Fieldname` with a Colon ":" in between. Globs are also supported to make it easier to obtain access to multiple resources or while developing. If you need access to all `Create` operations, you could use the following scopes: `"scopes": ["Mutation:Create*"]`
```

Comment:
out of interest can you do `"Mutation:*Person"` for all mutations related to people? This might be a bad idea if for example we have `MergeCompositionWithPerson`, does that also count (made up mutation)? Perhaps it's difficult to get correct.
Having said this, I'm pretty sure that almost all of our tokens will just be `"*"`, so perhaps this isn't worth spending time on.

Reply:
Yes, that would be possible and it will indeed also match mutations like `MergeCompositionWithPerson`. I've been thinking about this and we could add an extra colon to separate the type from the mutation. 

So it would look like this: `mutation:Person:Create`. This will make it easier to give full access to the Person type.

Reply:
That sounds like a good idea

Reply:
Ok, I'm on it